feat(tags): Add cards view for topics selection (desktop and tablet)

wanlingt opened the pull request:
## Problem

The views for topic selection are the same across all devices. On larger screens, this results in wasted space.

Closes #550 

## Solution

For tablets and desktops, use grids to display topics in individual boxes ([figma reference](https://www.figma.com/file/O6lxjjfn04Ow6nn2TMStpk/AskGov-Design-Master-v2?node-id=6003%3A70135))


## Before & After Screenshots

**BEFORE**:

https://user-images.githubusercontent.com/56983748/138998969-42407adc-e5f4-4bd0-8a63-25bf369f9725.mov


**AFTER**:

https://user-images.githubusercontent.com/56983748/138998873-c52d3903-1faa-44bb-a06c-3fb37302acf5.mov


## Tests

- Click around and check that navigation to topics still works

wanlingt committed:
feat(tagmenu): add cards view for topics on desktop and tablet

re #550

wanlingt committed:
feat(tagpanel): add cards view for topics on desktop and tablet

re #550

LoneRifle commented:
Have you considered the use of `useBreakpointValue()`[^1] to determine which grid to use, rather than multiple AccordionPanel definitions?

[^1]: https://chakra-ui.com/docs/hooks/use-breakpoint-value

wanlingt committed:
refactor(tags): use useBreakpointValue to control layout shown

re #550

wanlingt commented:
> Have you considered the use of `useBreakpointValue()`[1](#user-content-fn-1-6bf84e950f99ea5ca590b82bb06d9ceb) to determine which grid to use, rather than multiple AccordionPanel definitions?
> 
> ## Footnotes
> 1. https://chakra-ui.com/docs/hooks/use-breakpoint-value [↩](#user-content-fnref-1-6bf84e950f99ea5ca590b82bb06d9ceb)

just made some changes, something like that?

LoneRifle commented on client/src/components/TagMenu/TagMenu.component.tsx:
```diff
@@ -77,7 +80,114 @@ const TagMenu = (): ReactElement => {
+    sm: false,
+    md: false,
+    lg: false,
+    xl: false,
```
```suggestion
    base: true,
    sm: false,
```

LoneRifle commented on client/src/components/TagMenu/TagMenu.component.tsx:
```diff
@@ -77,7 +80,114 @@ const TagMenu = (): ReactElement => {
+          ({ tagType, tagname }) =>
+            tagType === TagType.Topic && tagname !== queryState,
+        )
+        .sort((a, b) => (a.tagname > b.tagname ? 1 : -1))
```
extract this to a `tagsToShow` const. In order to guarantee that `tagsToShow` is always an array of some form, consider something like...
```ts
const tagsToShow = (tags || []) // tags is possibly undefined, but if so, we default to an empty array
  .filter(...)
  .sort(...)
```

This then allows you to directly use `tagsToShow` without having to do a null check ie (`tags && `).

This also means that we will still render the VStack/SimpleGrid even if there are no tags to show, but I would regard this as a good thing since there are fewer changes to the DOM that are dependent on the application state

LoneRifle commented on client/src/components/TagMenu/TagMenu.component.tsx:
```diff
@@ -77,7 +80,114 @@ const TagMenu = (): ReactElement => {
+        )
+        .sort((a, b) => (a.tagname > b.tagname ? 1 : -1))
+        .map((tag) => {
+          const { tagType, tagname } = tag
```
```suggestion
        .map(({ id, tagType, tagname }) => {
```
then replace all references to tag with the appropriate property

LoneRifle reviewed (changes requested):
lgtm otherwise. Consider my points concerning the small rewrite and if you get stuck, feel free to talk to me

wanlingt committed:
refactor(tags): refactor tags into a separate const

- tagsToShow is used to represent tags or an empty array
- add bySpecifiedOrder to TagMenu

re #550

LoneRifle committed:
refactor(topics): ensure tagsToShow are filtered and sorted

LoneRifle reviewed (approved):
Merge after reviewing b6ed94c
